src/psyclone/tests/psyir/transformations/nemo_sign_test.py

```diff
@@ -0,0 +1,154 @@
+    assert trans._operator_name == "SIGN"
+    assert trans._classes == (BinaryOperation,)
+    assert trans._operators == (BinaryOperation.Operator.SIGN,)
+    # pylint: enable=protected-access
+    assert (str(trans) == "Convert the PSyIR SIGN intrinsic to equivalent "
+            "PSyIR code.")
+    assert trans.name == "NemoSignTrans"
+
+
+def example_psyir():
+    '''Utility function that creates a PSyIR tree containing a SIGN
+    intrinsic operator and returns the operator.
+
+    :returns: PSyIR SIGN operator instance.
+    :rtype: :py:class:`psyclone.psyGen.BinaryOperation`
+
+    '''
+    symbol_table = SymbolTable()
+    name1 = symbol_table.new_symbol_name("arg")
+    arg1 = DataSymbol(name1, DataType.REAL, interface=ArgumentInterface(
+        ArgumentInterface.Access.READWRITE))
+    symbol_table.add(arg1)
+    name2 = symbol_table.new_symbol_name("arg")
+    arg2 = DataSymbol(name2, DataType.REAL, interface=ArgumentInterface(
+        ArgumentInterface.Access.READWRITE))
+    symbol_table.add(arg2)
+    name3 = symbol_table.new_symbol_name()
+    symbol_table.add(DataSymbol(name3, DataType.REAL))
+    symbol_table.specify_argument_list([arg1, arg2])
+    var1 = Reference(name1)
+    var2 = Reference(name2)
+    var3 = Reference(name3)
+    oper = BinaryOperation.Operator.SIGN
+    operation = BinaryOperation.create(oper, var1, var2)
+    assign = Assignment.create(var3, operation)
+    _ = KernelSchedule.create("sign_example", symbol_table, [assign])
+    return operation
+
+
+def test_correct():
+    '''Check that a valid example produces the expected output.'''
+
+    Config.get().api = "nemo"
+    operation = example_psyir()
+    writer = FortranWriter()
+    result = writer(operation.root)
+    assert (
+        "subroutine sign_example(arg,arg_0)\n"
+        "  real, intent(inout) :: arg\n"
+        "  real, intent(inout) :: arg_0\n"
+        "  real :: psyir_tmp\n\n"
+        "  psyir_tmp=SIGN(arg, arg_0)\n\n"
+        "end subroutine sign_example\n") in result
+    trans = NemoSignTrans()
+    _, _ = trans.apply(operation, operation.root.symbol_table)
+    result = writer(operation.root)
+    assert (
+        "subroutine sign_example(arg,arg_0)\n"
+        "  real, intent(inout) :: arg\n"
+        "  real, intent(inout) :: arg_0\n"
+        "  real :: psyir_tmp\n"
+        "  real :: res_sign\n"
+        "  real :: tmp_sign\n"
+        "  real :: res_abs\n"
+        "  real :: tmp_abs\n\n"
+        "  tmp_abs=arg_0\n"
+        "  if (tmp_abs > 0.0) then\n"
+        "    res_abs=tmp_abs\n"
+        "  else\n"
+        "    res_abs=tmp_abs * -1.0\n"
+        "  end if\n"
+        "  res_sign=res_abs\n"
+        "  tmp_sign=arg\n"
+        "  if (tmp_sign < 0.0) then\n"
+        "    res_sign=res_sign * -1.0\n"
+        "  end if\n"
+        "  psyir_tmp=res_sign\n\n"
+        "end subroutine sign_example\n") in result
+    # Remove the created config instance
+    # pylint: disable=protected-access
```

Comment:
You shouldn't need these pylint directives within the test suite.

Reply:
Is there any harm in including them? It makes it easier to check for pylint compliance IMHO.

Reply:
Actually, what I meant by "shouldn't" was that pylint should not be complaining about them as they are in the test suite and I thought we had disabled this warning for the whole of the suite. Please could you investigate why this warning is not disabled.

Reply:
Investigation complete. The answer is that I have been running the test suite from src/psyclone, not src/psyclone/tests. This means I have not been picking up the .pylintrc file in tests. Seeing as the developer guide tells me to run the tests from the tests directory I consider this a 'user error' and have removed the directives.

Reply:
Great, thanks for getting to the bottom of it.